src/CompanyInfoGazette.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Worksome\CompanyInfo;
+
+use Illuminate\Http\Client\Response;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Http;
+
+class CompanyInfoGazette
+{
+    /**
+     * Lookup a company name on the English Gazette service, return processed company info.
+     *
+     * TODO: Should probably do a more specific query for the name, if possible?
+     *
+     * @param string $name Name of company to lookup.
+     *
+     * @return array|null Array of company info, or null if request to service failed.
+     */
+    public static function lookupName(string $name): ?array
```

Comment:
As this class actually has dependencies on the Http factory, let's instead make this method non-static. We can then add the Http factory to the constructor of this class and then we don't need to use the facade anymore. 

This would remove the coupling on facades to this class, and as facades has a dependency on the app container, it means we can use the CompanyInfoGazette without the need for an app container 👍

Reply:
Again, agreed. Let's steer clear of static functions. If we do want to be able to call these methods statically, we could have a facade for the new contract.